src/baldr/graphtile.cc

```diff
@@ -450,32 +428,29 @@ GraphTile::FileSuffix(const GraphId& graphid, const std::string& fname_suffix, b
                           : TileHierarchy::levels()[graphid.level()];
 
   // figure out how many digits
-  auto max_id = level.tiles.ncolumns() * level.tiles.nrows() - 1;
-  size_t max_length = static_cast<size_t>(std::log10(std::max(1, max_id))) + 1;
-  const size_t remainder = max_length % 3;
-  if (remainder) {
-    max_length += 3 - remainder;
-  }
+  auto get_num_digits = [&level]() {
+    auto max_id = level.tiles.ncolumns() * level.tiles.nrows() - 1;
+    size_t max_length = static_cast<size_t>(std::log10(std::max(1, max_id))) + 1;
+    const size_t remainder = max_length % 3;
+    if (remainder) {
+      max_length += 3 - remainder;
+    }
+    return max_length;
+  };
+  const size_t num_digits = get_num_digits();
+  const char separator = is_file_path ? filesystem::path::preferred_separator : '/';
 
-  // make a locale to use as a formatter for numbers
-  std::ostringstream stream;
-  if (is_file_path) {
-    stream.imbue(dir_locale);
-  } else {
-    stream.imbue(url_locale);
+  std::string tile_id_str(num_digits + num_digits / 3, '0');
+  assert(tile_id_str.size() % 4 == 0);
+  for (int sep_index = tile_id_str.size() - 4; sep_index >= 0; sep_index -= 4)
+    tile_id_str[sep_index] = separator;
+  size_t ind = tile_id_str.size();
+  for (auto tile_id = graphid.tileid(); tile_id != 0; tile_id /= 1000, ind -= 4) {
+    std::string tree_digits = std::to_string(tile_id % 1000);
+    std::copy(tree_digits.begin(), tree_digits.end(), tile_id_str.begin() + ind - tree_digits.size());
```

Comment:
Nit: This could be replaced with `snprintf` (which supports zero fill, e.g. with `"%03u"`)) and could also encompass printing the separator to the correct place to eliminate the previous for loop.

Reply:
I will rewrite it more optimal, without using other functions.